pkg/util/test/util.go

```diff
@@ -2578,6 +2610,255 @@ func ValidateAlertManagerDisabled(pxImageList map[string]string, cluster *corev1
 	return nil
 }
 
+// ValidateTelemetryEnabled validates telemetry component is running as expected
+func ValidateTelemetryEnabled(pxImageList map[string]string, cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	// Wait for the deployment to become online
+	logrus.Info("Validate px-telemetry-registration deployment and images")
+	registrationServiceDep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-telemetry-registration",
+			Namespace: cluster.Namespace,
+		},
+	}
+	if err := appops.Instance().ValidateDeployment(registrationServiceDep, timeout, interval); err != nil {
+		return err
+	}
+	registrationServiceDep, err := appops.Instance().GetDeployment(registrationServiceDep.Name, registrationServiceDep.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify registration container image inside px-telemetry-registration
+	telemetryImageName, ok := pxImageList["telemetry"]
+	if !ok {
+		return fmt.Errorf("failed to find image for telemetry")
+	}
+	telemetryImageName = util.GetImageURN(cluster, telemetryImageName)
+
+	if registrationServiceDep.Spec.Template.Spec.Containers[0].Image != telemetryImageName {
+		return fmt.Errorf("registration image mismatch, image: %s, expected: %s",
+			registrationServiceDep.Spec.Template.Spec.Containers[0].Image,
+			telemetryImageName)
+	}
+	// Verify envoy container image inside registration-service
+	envoyImageName, ok := pxImageList["telemetryProxy"]
+	if !ok {
+		return fmt.Errorf("failed to find image for envoy")
+	}
+	envoyImageName = util.GetImageURN(cluster, envoyImageName)
+	if registrationServiceDep.Spec.Template.Spec.Containers[1].Image != envoyImageName {
+		return fmt.Errorf("envoy image mismatch, image: %s, expected: %s",
+			registrationServiceDep.Spec.Template.Spec.Containers[1].Image,
+			envoyImageName)
+	}
+
+	// Verify px-telemetry-metrics-collector deployment and images
```

Comment:
can we validate the deployment has a init container as well?

Reply:
Done